sdk/python/kfp/deprecated/_client.py

```diff
@@ -1161,6 +1162,167 @@ def enable_job(self, job_id: str):
+          interval_second: Integer indicating the seconds between two recurring runs in for a periodic schedule.
+          cron_expression: A cron expression representing a set of times, using 6 space-separated fields, e.g. "0 0 9 ? * 2-6".
+            See `here <https://pkg.go.dev/github.com/robfig/cron#hdr-CRON_Expression_Format>`_ for details of the cron expression format.
+          max_concurrency: Integer indicating how many jobs can be run in parallel.
+          no_catchup: Whether the recurring run should catch up if behind schedule.
+            For example, if the recurring run is paused for a while and re-enabled
+            afterwards. If no_catchup=False, the scheduler will catch up on (backfill) each
+            missed interval. Otherwise, it only schedules the latest interval if more than one interval
+            is ready to be scheduled.
+            Usually, if your pipeline handles backfill internally, you should turn catchup
+            off to avoid duplicate backfill. (default: {False})
+          pipeline_package_path: Local path of the pipeline package(the filename should end with one of the following .tar.gz, .tgz, .zip, .yaml, .yml).
+          params: A dictionary with key (string) as param name and value (string) as param value.
+          pipeline_id: The id of a pipeline.
+          version_id: The id of a pipeline version.
+            If both pipeline_id and version_id are specified, version_id will take precendence.
+            If only pipeline_id is specified, the default version of this pipeline is used to create the run.
+          enabled: A bool indicating whether the recurring run is enabled or disabled.
+          enable_caching: Optional. Whether or not to enable caching for the run.
+            This setting affects v2 compatible mode and v2 mode only.
+            If not set, defaults to the compile time settings, which are True for all
+            tasks by default, while users may specify different caching options for
+            individual tasks.
+            If set, the setting applies to all tasks in the pipeline -- overrides
+            the compile time settings.
+          service_account: Optional. Specifies which Kubernetes service account this
+            recurring run uses.
+
+        Returns:
+          A Job object. Most important field is id.
+
+        Raises:
+          ValueError: If required parameters are not supplied.
+        """
+        job = self.get_recurring_run(job_id=job_id)
+
+        # We iterate through the input parameters and determine whether to use
+        # the corresponding values in the retrieved job or consume the override
+        # passed to this function.
+
+        if not experiment_id:
+            for resource_reference in job.resource_references:
+                if resource_reference.relationship == "OWNER":
+                    experiment_id = resource_reference.key.id
+                    break
+
+        if not job_name:
```

Comment:
Consider using the following assignment expression throughout where possible for simplicity:
```python
job_name = job_name or f"Clone of {job.name}"
```